Simplify Core

f2prateek opened the pull request:
Worked on this a while back, feel free to close this if you already have something in the works. Ref #172.

f2prateek commented:
Realized I had reformatted the structure, let me know if you want me to change it back.

JakeWharton commented:
As painful as it is, we probably need to keep the four    space    indent    to    be    consistent.

Also per the linked issue we need to do a final deprecate-all-the-things release before we'd merge.

f2prateek commented:
Rebased and reverted the formatting to the original. I tried to only delete things as much as I could, and not fix other issues I noticed (like formatting https://cloudup.com/cmAg0ka9yxE).

JakeWharton commented on sample-app/src/main/java/rx/android/samples/MainActivity.java:
```diff
@@ -0,0 +1,73 @@
+            }
+        }) //
+                .subscribeOn(handlerThread(backgroundThreadHandler)) // Run it on a background thread
+                .observeOn(AndroidSchedulers.mainThread())           // Be notified on the main thread
```
Either static import `mainThread()` or un-static import `handlerThread()`.

JakeWharton commented:
Awesome. LGTM. Waiting on #184.
